core/drivers/crypto/se050/core/ecc.c

```diff
@@ -0,0 +1,630 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (C) Foundries Ltd. 2020 - All Rights Reserved
+ * Author: Jorge Ramirez <jorge@foundries.io>
+ */
+
+#include <assert.h>
+#include <der.h>
+#include <drvcrypt.h>
+#include <drvcrypt_acipher.h>
+#include <drvcrypt_math.h>
+#include <initcall.h>
+#include <se050.h>
+#include <string.h>
+#include <tee/cache.h>
+#include <tee/tee_cryp_utl.h>
+#include <tee_api_defines_extensions.h>
+#include <utee_defines.h>
+#include <util.h>
+
+static uint32_t algo_tee2se050(uint32_t algo)
+{
+	switch (algo) {
+	case TEE_ALG_ECDSA_P192:
+		return kAlgorithm_SSS_ECDSA_SHA1;
+	case TEE_ALG_ECDSA_P224:
+		return kAlgorithm_SSS_ECDSA_SHA224;
+	case TEE_ALG_ECDSA_P256:
+		return kAlgorithm_SSS_ECDSA_SHA256;
+	case TEE_ALG_ECDSA_P384:
+		return kAlgorithm_SSS_ECDSA_SHA384;
+	case TEE_ALG_ECDSA_P521:
+		return kAlgorithm_SSS_ECDSA_SHA512;
+	default:
+		EMSG("ecc curve 0x%x not enabled", algo);
+		return kAlgorithm_None;
+	}
+}
+
+static uint32_t cipher_tee2se050(uint32_t curve)
+{
+	switch (curve) {
+	case TEE_ECC_CURVE_NIST_P192:
+	case TEE_ECC_CURVE_NIST_P224:
+	case TEE_ECC_CURVE_NIST_P256:
+	case TEE_ECC_CURVE_NIST_P384:
+	case TEE_ECC_CURVE_NIST_P521:
+		return kSSS_CipherType_EC_NIST_P;
+	default:
+		EMSG("cipher 0x%x not enabled", curve);
```

Comment:
curve is 3bit, should use `"%"PRIx32` => `EMSG("cipher %#"PRIx32" not enabled", curve);`
Ditto in function above and 4 other occurences below.

Reply:
yes.
what about doing?
`EMSG("cipher 0x%" PRIx32 " not enabled", curve);
`
seems to be the norm in other parts of OP-TEE?

Reply:
In the last years we've started to prefer the form suggested by Etienne, it's a bit more compact and also easier to read in my opinion.

Reply:
ok